### PR TITLE
Add interface check to db & censustree impl

### PR DESCRIPTION
**Pull request by arnaucube**
Adds interface checks to ensure that the different implementations of
`db.Database` & `censustree.Tree` match the interfaces.

### CODE REVIEW

File: censustree/gravitontree/trie_test.go [hunk 1]
```diff
@@ -6,9 +6,25 @@ import (
 	"testing"
 	"time"
 
+	"go.vocdoni.io/dvote/censustree"
 	"go.vocdoni.io/dvote/util"
 )
 
+func TestInterface(t *testing.T) {
+	storage := t.TempDir()
```
Comment (p4u) [inline censustree/gravitontree/trie_test.go]: no need to declare storage as a var

File: censustree/gravitontree/trie_test.go [hunk 2]
```diff
@@ -6,9 +6,25 @@ import (
...
+	tree := &Tree{}
+	err := tree.Init("test", storage)
+	if err != nil {
```
Comment (p4u) [inline censustree/gravitontree/trie_test.go]: please, use syntax `if err := tree.Init()...` when possible

File: censustree/gravitontree/trie_test.go [hunk 3]
```diff
@@ -6,9 +6,25 @@ import (
...
+		t.Fatal(err)
+	}
+
+	var i interface{} = tree
+	_, ok := i.(censustree.Tree)
```
Comment (p4u) [inline censustree/gravitontree/trie_test.go]: same, use inline `if` declaration

**Review by p4u — CHANGES_REQUESTED** [review]

**Review by p4u — APPROVED** [review]

**Comment by mvdan** [comment]
Why do we need actual tests here, though? Usually one just does `var _ SomeInterface = (*SomeImplementation)(nil)` as a static assertion that SomeImplementation implements SomeInterface. The package will stop building if the assertion is false.

**Comment by arnaucube** [comment]
> Why do we need actual tests here, though? Usually one just does `var _ SomeInterface = (*SomeImplementation)(nil)` as a static assertion that SomeImplementation implements SomeInterface. The package will stop building if the assertion is false.

True! Amended the commit changing the added tests with the test that you've mentioned.
The motivation to add these tests is that if in the future the implementation is changed it can be automatically detected if it breaks the matching with the interface. As an example, the current [db.Iden3Storage](https://github.com/vocdoni/vocdoni-node/blob/master/db/iden3.go) does not match the `db.Database` interface, but as there is no test checking it, is not detected.

**Review by mvdan — APPROVED** [review]

**Comment by arnaucube** [comment]
After @mvdan comment and chat, amended the commit to move the checks from test files to the actual implementation files. Much better in this way ^^
